app/controllers/assignments_controller.rb

```diff
@@ -5,6 +5,7 @@ class AssignmentsController < ApplicationController
 
   protect_from_forgery with: :exception
 
+  before_action :authenticate_user!, only:[:new, :create, :edit, :update]
```

Comment:
This should also be a `SecureController` but it just skips the authentication on `:index`.